R/seq_linter.R

```diff
@@ -16,6 +16,16 @@ seq_linter <- function() {
     and expr[expr[(expr|self::*)[SYMBOL_FUNCTION_CALL[ {xp_text_in_table(bad_funcs)} ]]]]
   ]")
 
+  ## The actual order of the nodes is document order
+  ## In practice we need to handle length(x):1
+  get_fun <- function(expr, n) {
+    funcall <- xml2::xml_find_chr(expr, sprintf("string(./expr[%d])", n))
+    fun <- gsub("\\(.*\\)", "(...)", trimws(funcall))
```

Comment:
Is this `xp_call_name(expr, depth  = n)` with a `(...)` suffix?

Reply:
not quite, no... the call we're at is `:`(x, y) in infix form, so call logic doesn't quite apply. and it only applies to one half of the expr (`1:length(x)` or `length(x):1`)...

i recall trying to simplify this previously and it being a bit of a pain. willing to revisit again later but for now focused on vectorizing as straightforwardly as possible.